File: trail2/숫자 카운트/numeric-count.py
n = int(input())
a, b, c = [], [], []
for _ in range(n):
    num, cnt1, cnt2 = map(int, input().split())
    a.append(str(num))
    b.append(cnt1)
    c.append(cnt2)

# Please write your code here.
ans = 0

for i in range(1, 10):
    for j in range(1, 10):
        
        if i == j:
            continue

        for k in range(1, 10):

            if j == k or i == k:
                continue

            t = str(i) + str(j) + str(k)
            f = True

            for x in range(n):
                c1 = 0
                c2 = 0

                g = a[x]

                for y in range(3):
                    if t[y] == g[y]:
                        c1 += 1

                    elif g[y] in t:
                        c2 += 1

                if c1 != b[x] or c2 != c[x]:
                    f = False
                    break

            if f:
                ans += 1

print(ans)

File: trail2/숫자 카운트/numeric-count.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] num = new int[n];
        int[] count1 = new int[n];
        int[] count2 = new int[n];
        for (int i = 0; i < n; i++) {
            num[i] = sc.nextInt();
            count1[i] = sc.nextInt();
            count2[i] = sc.nextInt();
        }
        // Please write your code here.

        int ans = 0;

        for (int i = 1; i < 10; i++) {
            for (int j = 1; j < 10; j++) {

                if (i == j) {
                    continue;
                }

                for (int k = 1; k < 10; k++) {

                    if (i == k || j == k) {
                        continue;
                    }

                    String t = "" + i + j + k;
                    boolean f = true;

                    for (int x = 0; x < n; x++) {
                        int c1 = 0;
                        int c2 = 0;

                        String g = "" + num[x];

                        for (int y = 0; y < 3; y++) {
                            
                            if (g.charAt(y) == t.charAt(y)) {
                                c1++;
                            }
                            else if (t.contains(String.valueOf(g.charAt(y)))) {
                                c2++;
                            }
                        }

                        if (count1[x] != c1 || count2[x] != c2) {
                            f = false;
                            break;
                        }
                    }
                    
                    if (f) {
                        ans++;
                    }

                }
            }
        }
        System.out.println(ans);
    }
}